{
  "active_lock_reason": null,
  "assignee": {
    "avatar_url": "https://avatars3.githubusercontent.com/u/1944093?v=4",
    "events_url": "https://api.github.com/users/caseydavenport/events{/privacy}",
    "followers_url": "https://api.github.com/users/caseydavenport/followers",
    "following_url": "https://api.github.com/users/caseydavenport/following{/other_user}",
    "gists_url": "https://api.github.com/users/caseydavenport/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/caseydavenport",
    "id": 1944093,
    "login": "caseydavenport",
    "node_id": "MDQ6VXNlcjE5NDQwOTM=",
    "organizations_url": "https://api.github.com/users/caseydavenport/orgs",
    "received_events_url": "https://api.github.com/users/caseydavenport/received_events",
    "repos_url": "https://api.github.com/users/caseydavenport/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/caseydavenport/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/caseydavenport/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/caseydavenport"
  },
  "assignees": [
    {
      "avatar_url": "https://avatars3.githubusercontent.com/u/1944093?v=4",
      "events_url": "https://api.github.com/users/caseydavenport/events{/privacy}",
      "followers_url": "https://api.github.com/users/caseydavenport/followers",
      "following_url": "https://api.github.com/users/caseydavenport/following{/other_user}",
      "gists_url": "https://api.github.com/users/caseydavenport/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/caseydavenport",
      "id": 1944093,
      "login": "caseydavenport",
      "node_id": "MDQ6VXNlcjE5NDQwOTM=",
      "organizations_url": "https://api.github.com/users/caseydavenport/orgs",
      "received_events_url": "https://api.github.com/users/caseydavenport/received_events",
      "repos_url": "https://api.github.com/users/caseydavenport/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/caseydavenport/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/caseydavenport/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/caseydavenport"
    }
  ],
  "author_association": "CONTRIBUTOR",
  "body": "**What happened**:\r\n\r\nWhen building on `master` sync fail messages are observed in the `kube-proxy` log when mode=ipvs is used;\r\n\r\n```\r\nE0723 07:41:15.295733     243 proxier.go:1234] Failed to execute iptables-restore: exit status 2 (iptables-restore v1.8.2 (legacy): Couldn't load target `KUBE-MARK-DROP':No such file or directory\r\n\r\nError occurred at line: 12\r\nTry `iptables-restore -h' or 'iptables-restore --help' for more information.\r\n)\r\nRules:\r\n*nat\r\n:KUBE-POSTROUTING - [0:0]\r\n:KUBE-POSTROUTING - [0:0]\r\n:KUBE-POSTROUTING - [0:0]\r\n:KUBE-POSTROUTING - [0:0]\r\n:KUBE-POSTROUTING - [0:0]\r\n:KUBE-MARK-MASQ - [0:0]\r\n-A KUBE-POSTROUTING -m comment --comment \"kubernetes service traffic requiring SNAT\" -m mark --mark 0x00004000/0x00004000 -j MASQUERADE\r\n-A KUBE-MARK-MASQ -j MARK --set-xmark 0x00004000/0x00004000\r\n-A KUBE-SERVICES -m addrtype --dst-type LOCAL -j KUBE-NODE-PORT\r\n-A KUBE-LOAD-BALANCER -j KUBE-MARK-MASQ\r\n-A KUBE-FIREWALL -j KUBE-MARK-DROP\r\nCOMMIT\r\n*filter\r\n:KUBE-FORWARD - [0:0]\r\n-A KUBE-FORWARD -m comment --comment \"kubernetes forwarding rules\" -m mark --mark 0x00004000/0x00004000 -j ACCEPT\r\n-A KUBE-FORWARD -s 11.0.0.0/16 -m comment --comment \"kubernetes forwarding conntrack pod source rule\" -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT\r\n-A KUBE-FORWARD -m comment --comment \"kubernetes forwarding conntrack pod destination rule\" -d 11.0.0.0/16 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT\r\nCOMMIT\r\n```\r\n\r\nThe `KUBE-MARK-DROP` chain is not defined (but `KUBE-POSTROUTING` is defined 5 times)\r\n\r\n**What you expected to happen**:\r\n\r\nSync shall succeed. The chain `KUBE-MARK-DROP` shall be defined before use in the \"restore\" command to iptables.\r\n\r\n**How to reproduce it (as minimally and precisely as possible)**:\r\n\r\n* Start a cluster built on master with proxy-mode=ipvs\r\n* Check the kube-proxy logs\r\n\r\n**Anything else we need to know?**:\r\n\r\nAll tests works fine. Also this may be a fault during start-up, I can se only one fault logging. Anyway this log printout does not exist in 1.15.\r\n\r\n**Environment**:\r\n- Kubernetes version (use `kubectl version`):\r\n```\r\nClient Version: version.Info{Major:\"1\", Minor:\"16+\", GitVersion:\"v1.16.0-alpha.1.151+7dc5d1f0646e0d\", GitCommit:\"7dc5d1f0646e0db0c077db578cf4b0855d8c799a\", GitTreeState:\"clean\", BuildDate:\"2019-07-23T07:38:45Z\", GoVersion:\"go1.12.5\", Compiler:\"gc\", Platform:\"linux/amd64\"}\r\nServer Version: version.Info{Major:\"1\", Minor:\"16+\", GitVersion:\"v1.16.0-alpha.1.151+7dc5d1f0646e0d\", GitCommit:\"7dc5d1f0646e0db0c077db578cf4b0855d8c799a\", GitTreeState:\"clean\", BuildDate:\"2019-07-23T07:38:14Z\", GoVersion:\"go1.12.5\", Compiler:\"gc\", Platform:\"linux/amd64\"}\r\n```\r\n\r\n- Cloud provider or hardware configuration:\r\n- OS (e.g: `cat /etc/os-release`):\r\n- Kernel (e.g. `uname -a`):\r\n- Install tools:\r\n- Network plugin and version (if this is a network-related bug):\r\n- Others:\r\n",
  "closed_at": "2019-09-03T23:01:38Z",
  "closed_by": {
    "avatar_url": "https://avatars0.githubusercontent.com/u/20407524?v=4",
    "events_url": "https://api.github.com/users/k8s-ci-robot/events{/privacy}",
    "followers_url": "https://api.github.com/users/k8s-ci-robot/followers",
    "following_url": "https://api.github.com/users/k8s-ci-robot/following{/other_user}",
    "gists_url": "https://api.github.com/users/k8s-ci-robot/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/k8s-ci-robot",
    "id": 20407524,
    "login": "k8s-ci-robot",
    "node_id": "MDQ6VXNlcjIwNDA3NTI0",
    "organizations_url": "https://api.github.com/users/k8s-ci-robot/orgs",
    "received_events_url": "https://api.github.com/users/k8s-ci-robot/received_events",
    "repos_url": "https://api.github.com/users/k8s-ci-robot/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/k8s-ci-robot/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/k8s-ci-robot/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/k8s-ci-robot"
  },
  "comments": 14,
  "comments_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/80462/comments",
  "created_at": "2019-07-23T08:04:53Z",
  "events_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/80462/events",
  "html_url": "https://github.com/kubernetes/kubernetes/issues/80462",
  "id": 471544754,
  "labels": [
    {
      "color": "0052cc",
      "default": false,
      "description": null,
      "id": 755527763,
      "name": "area/ipvs",
      "node_id": "MDU6TGFiZWw3NTU1Mjc3NjM=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/area/ipvs"
    },
    {
      "color": "e11d21",
      "default": false,
      "description": "Categorizes issue or PR as related to a bug.",
      "id": 105146071,
      "name": "kind/bug",
      "node_id": "MDU6TGFiZWwxMDUxNDYwNzE=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/kind/bug"
    },
    {
      "color": "d2b48c",
      "default": false,
      "description": "Categorizes an issue or PR as relevant to SIG Network.",
      "id": 116712108,
      "name": "sig/network",
      "node_id": "MDU6TGFiZWwxMTY3MTIxMDg=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/sig/network"
    }
  ],
  "labels_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/80462/labels{/name}",
  "locked": false,
  "milestone": null,
  "node_id": "MDU6SXNzdWU0NzE1NDQ3NTQ=",
  "number": 80462,
  "performed_via_github_app": null,
  "repository_url": "https://api.github.com/repos/kubernetes/kubernetes",
  "state": "closed",
  "title": "Kube-proxy/ipvs; sync logs fault on master; KUBE-MARK-DROP missing",
  "updated_at": "2019-09-03T23:01:38Z",
  "url": "https://api.github.com/repos/kubernetes/kubernetes/issues/80462",
  "user": {
    "avatar_url": "https://avatars1.githubusercontent.com/u/37046727?v=4",
    "events_url": "https://api.github.com/users/uablrek/events{/privacy}",
    "followers_url": "https://api.github.com/users/uablrek/followers",
    "following_url": "https://api.github.com/users/uablrek/following{/other_user}",
    "gists_url": "https://api.github.com/users/uablrek/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/uablrek",
    "id": 37046727,
    "login": "uablrek",
    "node_id": "MDQ6VXNlcjM3MDQ2NzI3",
    "organizations_url": "https://api.github.com/users/uablrek/orgs",
    "received_events_url": "https://api.github.com/users/uablrek/received_events",
    "repos_url": "https://api.github.com/users/uablrek/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/uablrek/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/uablrek/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/uablrek"
  }
}